src/main/java/com/minecolonies/colony/ColonyManager.java

```diff
@@ -87,6 +93,53 @@ public static Colony createColony(@NotNull World w, BlockPos pos, @NotNull Entit
     }
 
     /**
+     * Delete a colony and kill all citizens/purge all buildings.
+     *
+     * @param id the colonies id
+     */
+    public static void deleteColony(int id)
+    {
+        try
+        {
+            final Colony colony = getColony(id);
+            Log.getLogger().info("Deleting colony " + id);
+            colonies.remove(id);
+            coloniesByWorld.get(colony.getDimension()).remove(colony);
+            final Set<World> colonyWorlds = new HashSet<>();
+            Log.getLogger().info("Removing citizens for " + id);
+            for (final CitizenData citizenData : new ArrayList<>(colony.getCitizens().values()))
+            {
+                Log.getLogger().info("Kill Citizen " + citizenData.getName());
+                World world = citizenData.getCitizenEntity().getEntityWorld();
+                citizenData.getCitizenEntity().onDeath(CONSOLE_DAMAGE_SOURCE);
+                colonyWorlds.add(world);
+            }
+            Log.getLogger().info("Removing buildings for " + id);
+            for (final AbstractBuilding building : new ArrayList<>(colony.getBuildings().values()))
+            {
+
+                final BlockPos location = building.getLocation();
+                Log.getLogger().info("Delete Building at " + location);
+                building.destroy();
+                for (final World world : colonyWorlds)
```

Comment:
And I check that it is a hut block before setting it, so it isn't really dangerous. Worst case is we miss a block.